### DISCUSSION TITLE
Running --watch on linux results in "error: ModuleNotFound resolving "index.js" (entry point)"

### DISCUSSION THREAD

**Discussion by mhanberg** [oven-sh/bun#14591] [Q&A] [answered]
Hi, as the title says, when I run bun in watch mode on linux, I am getting the error `error: ModuleNotFound resolving "index.js" (entry point)`, which requires restarting bun. This happens after a few file changes.

When running on macos however, I am not seeing this at all, it just works.

Any ideas?

The command is `bun build index.js --outdir=../_site/js` and i have the following package.json

```json
{
  "name": "site",
  "main": "index.js",
  "devDependencies": {
    "@types/bun": "latest"
  },
  "dependencies": {
    "fzf": "^0.5.2",
    "lit-html": "^3.2.1"
  }
}
```

**Comment by mhanberg** [comment]
I'm actually wondering if this is related to saving the file in vim and how vim flushes files to disk...

**Reply by Jarred-Sumner** [reply]
it very likely is related to that. It's possible that the watcher loaded faster than vim could finish renaming the file in-place

**Reply by mhanberg** [reply, marked as answer]
I forgot to come back, this is directly related to: https://github.com/oven-sh/bun/issues/2987#issuecomment-2403940233

in vim, setting `set noswapfile` fixed it, as i believe that makes vim just write it to disk instead of doing a dance with touching the file multiple times

**Comment by drkwolf** [comment]
This is due to the file write handling of Vim. See :help 'backupcopy' for an explanation. Editors like Vim replace the original file with a temporary backup to avoid losing the file contents completely. This is also an issue when watching the file for changes via inotifywait ([see here](https://stackoverflow.com/questions/10507344/get-vim-to-modify-the-file-instead-of-moving-the-new-version-on-it)). A workaround for that is to
`
:set backupcopy=yes
`
You'll still see events for the backup file, but at least it's for another file. To completely forego the safety of a backup, you can addditionally
`
:set nobackup
`
